R/tax_range_strat.R

```diff
@@ -102,20 +106,31 @@
 #'                           color = c("#67C5CA", "#F2F91D"))
 #' axis_geo(side = 4, intervals = eras_custom, tick_labels = FALSE)
 #' title(xlab = "Taxon", line = 10.5)
+#' # Pull class data
+#' occdf$class <- tetrapods$class[1:50]
+#' # Group stratigraphic ranges by class
+#' tax_range_strat(occdf, name = "taxon", group = "class",
+#'                 certainty = "certainty", by = "name",
+#'                 plot_args = list(main = "Section A",
+#'                                  ylab = "Stratigraphic height (m)"))
 #'
 #' @export
-tax_range_strat <- function(occdf, name = "genus", level = "bed",
+tax_range_strat <- function(occdf, name = "genus", level = "bed", group = NULL,
                             certainty = NULL, by = "FAD", plot_args = NULL,
                             x_args = NULL, y_args = NULL) {
 
   if (is.data.frame(occdf) == FALSE) {
-    stop("`occdf` should be a dataframe")
+    stop("`occdf` should be a data.frame")
   }
 
   if (!is.numeric(occdf[, level, drop = TRUE])) {
     stop("`level` must be of class numeric")
   }
 
+  if (!is.null(group) && (group %in% colnames(occdf) == FALSE)) {
+    stop('`group` is not a named column in `occdf`')
+  }
```

Comment:
As @bethany-j-allen mentioned, this line breaks if more than one grouping column is supplied (e.g., `group = c("class", "order")`. I think it's fine to only allow a single grouping variable, but we should do a check to make sure that's a case.

Reply:
Same for the same code in `tax_range_time.R`